### PR TITLE
Energy blade rewrite, organ qdel cleanup.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Cleans up energy swords and some associated mob code. 
Also lazylists some tangentially related vars because I was in the area.
Also also fixes some incidental bugs/runtimes noticed during testing.

## Why and what will this PR improve
- Another step on the road to killing `item_state`.
- Less horrible ninja blade override checking, checks the base type instead now.
- More flexible overlay-based energy sword code that uses the single icon system.
- Lowers memory footprint and init time of /mob by lazylisting a couple of vars.
- May fix some GC problems with human organs, embedded and pinning items.

## Authorship
Me.

## Changelog
:cl:
tweak: All energy weapons (ninja blade, esword, energy axe, energy cutlass, energy machete) can be used as scalpels when they are energized.
tweak: Energy swords can be used to slice open safes, lockers, windoors and tables a la the ninja blade when they are energized.
/:cl:

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Ok I need to stop here or I'm going to keep lazylisting stuff and increasing the diff.

**Comment by MistakeNot4892** [comment]
- Pinning, embedding, organ removal, embedded item removal are all working.
- Energy weapons are ~~more or less~~ working ~~other than a layering issue with the active inhand icon.~~ Icon layering fixed.
- ~~Turning on an esword crashes DD instantly so that's cool.~~ Hasn't happened since restarting, no idea what the cause was.
- Overlays are not being applied to inhands when held by neo-avians.
- ~~Esword deactivates when swapping hands.~~
- ~~Ninja blade hasn't been tested.~~

**Comment by MistakeNot4892** [comment]
Last visible issue is overlays not working for nonhumans and that seems to be a deeper problem than this PR so calling it done.

File: code/game/objects/items/weapons/melee/energy_ninja.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	active_throwforce =   1  //Throwing or dropping the item deletes it.
+	active_parry_chance = 50
+
+	var/mob/living/creator
+
+/obj/item/energy_blade/ninja/Initialize()
+	. = ..()
+	if(!ismob(loc))
+		return INITIALIZE_HINT_QDEL
+	START_PROCESSING(SSobj, src)
```
Comment (comma) [inline code/game/objects/items/weapons/melee/energy_ninja.dm]: wots this for

Comment (MistakeNot4892) [reply code/game/objects/items/weapons/melee/energy_ninja.dm]: Artifact I think.

**Review by comma — APPROVED** [review]

**Pull request merged**